app/src/main/java/org/greenstand/android/TreeTracker/signup/SignupViewModel.kt

```diff
@@ -7,26 +7,55 @@ import org.greenstand.android.TreeTracker.models.Users
 
 // Dequeue breaks equals so state will not be updated when navigating
 data class SignUpState(
-    val emailPhone: String? = null,
+    val emailText: String = "",
+    val phoneText: String = "",
     val name: String? = null,
     val photoPath: String? = null,
+    val credentialType: CredentialType = CredentialType.Email,
+    val showEmailText: Boolean = true,
+    val showPhoneText: Boolean = false,
```

Comment:
I take it if one is false, the other is true. In that case we only need one boolean.